app/scripts/modules/core/src/delivery/details/StageFailureMessage.tsx

```diff
@@ -76,21 +94,38 @@ export class StageFailureMessage extends React.Component<IStageFailureMessagePro
         );
       }
 
-      const currentState = ReactInjector.$state.current;
-      return (
-        <div className="row">
-          <div className="col-md-12">
-            <div className="alert alert-danger">
-              <div>
-                Stage <UISref to={currentState.name} params={{ step: failedStageIndex }}>
+      if (failedStage && (failedStageIndex || failedStepIndex)) {
+        const currentState = ReactInjector.$state.current;
+        const params: any = {
+          'step': failedStepIndex
+        };
+        if (failedExecutionId) {
+          params.executionId = failedExecutionId;
+        }
+        if (failedStageIndex) {
+          params.stage = failedStageIndex;
+        }
```

Comment:
any reason not to just do:
```javascript
const params: any = {
  step: failedStepIndex,
  executionId: failedExecutionId,
  stage: failedStageIndex,
};
```
?

(Not 100% sure of the ui-router behavior but  would clean this up quite a bit)

Reply:
I did try this (and just tried it again), feels like the router gets all messed up with an undefined `executionId`. 

Did not dig farther than seeing that the links did not work when the executionId was not provided.

I didn't see the `executionId` on `IExecutionStage` itself otherwise I could always set it and clean this up.

Is there some other way to grab the executionId for the stage? Looked like the ReactInjector but couldn't immediately find a way to pull it from `$state.current`.